lib/torrent.js

```diff
@@ -638,24 +639,70 @@ export default class Torrent extends EventEmitter {
     })
   }
 
-  _verifyPieces (cb) {
-    parallelLimit(this.pieces.map((piece, index) => cb => {
+  _verifyPiece (index, cb) {
+    if (this.destroyed) return cb(new Error('torrent is destroyed'))
+
+    const getOpts = {}
+    // Specify length for the last piece in case it is zero-padded
+    if (index === this.pieces.length - 1) {
+      getOpts.length = this.lastPieceLength
+    }
+    this.store.get(index, getOpts, async (err, buf) => {
       if (this.destroyed) return cb(new Error('torrent is destroyed'))
 
-      const getOpts = {}
-      // Specify length for the last piece in case it is zero-padded
-      if (index === this.pieces.length - 1) {
-        getOpts.length = this.lastPieceLength
-      }
-      this.store.get(index, getOpts, async (err, buf) => {
-        if (this.destroyed) return cb(new Error('torrent is destroyed'))
+      if (err) return queueMicrotask(() => cb(null)) // ignore error
+
+      const hex = await hash(buf, 'hex')
+      if (this.destroyed) return cb(new Error('torrent is destroyed'))
 
-        if (err) return queueMicrotask(() => cb(null)) // ignore error
+      cb(null, hex === this._hashes[index])
+    })
+  }
 
-        const hex = await hash(buf, 'hex')
-        if (this.destroyed) return cb(new Error('torrent is destroyed'))
+  _verifyPiecesUsingBitfield (cb) {
+    const piecesToCheck = new Set()
+    this.files.forEach(file => {
+      let checkFile = true
+      for (let i = file._startPiece; i <= file._endPiece; ++i) {
+        if (this._startupBitfield.get(i)) {
+          if (checkFile) {
+            piecesToCheck.add(i)
+            checkFile = false
```

Comment:
shouldn't this have a break? I think the point of this is to look if any of the bitfield pieces are in the file's range, once we find one we don't need to continue iterating? that allows u to get rid of `checkFile`

Reply:
My bad, after re-reading the code we don't want a break there because it needs to continue to mark the rest of the pieces of the file as valid or no

Reply:
oh right, i see, i'm stupid, I thought you did this differently, you copy the contents of the startup bitfield into the torrent's bitfield, imho the torrent's bitfield should be the startup bitfield, and if something goes wrong the pieces from it should be invalidated, this is also why i think the initial bitfield should be a buffer, the developer doesnt really care about the bitfield class when he wants to store it, just it's buffer